File: advanced-programming-methods/labs/toy-interpreter-java/src/model/IStack.java
package model;

import exceptions.StackException;
import model.statement.Statement;

/**
 * The interface I stack.
 */
public interface IStack extends java.io.Serializable
{
    /**
     * Top statement.
     *
     * @return the statement
     * @throws StackException the stack exception
     */
    public Statement top() throws StackException;

    /**
     * Pop statement.
     *
     * @return the statement
     * @throws StackException the stack exception
     */
    public Statement pop() throws StackException;

    /**
     * Getter for saving the stack into a file
     *
     * @return String
     */
    public String getAllSave();

    /**
     * Push void.
     *
     * @param element the element
     * @throws StackException the stack exception
     */
    public void push(Statement element) throws StackException;

    public void clear();

    /**
     * Is empty.
     *
     * @return the boolean
     */
    public boolean isEmpty();
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/IStack.cs
using Exceptions;

namespace Model
{
    /// <summary>
    /// The interface I stack.
    /// </summary>
    public interface IStack
    {
        /// <summary>
        /// Top statement.
        /// </summary>
        /// <returns> the statement </returns>
        /// <exception cref="StackException"> the stack exception </exception>
        Statement Top();

        /// <summary>
        /// Pop statement.
        /// </summary>
        /// <returns> the statement </returns>
        /// <exception cref="StackException"> the stack exception </exception>
        Statement Pop();

        /// <summary>
        /// Push void.
        /// </summary>
        /// <param name="element"> the element </param>
        /// <exception cref="StackException"> the stack exception </exception>
        void Push(Statement element);

        void Clear();

        /// <summary>
        /// Getter for saving the stack into a file
        /// </summary>
        /// <value>All save.</value>
        string AllSave {get;}

        /// <summary>
        /// Is empty.
        /// </summary>
        /// <returns> the boolean </returns>
        bool Empty {get;}
    }
}
    